packages/react/src/core/CoreProvider.tsx

```diff
@@ -26,6 +26,8 @@ export type LinkProps = PropsWithChildren<{
 	download?: NativeLinkProps['download'];
 	/** The URL that the hyperlink points to. */
 	href?: string;
+	/** Function to be fired following a click event of the link. */
+	onClick?: NativeLinkProps['onClick'];
```

Comment:
The `onClick` prop has technically always worked, but we've never added support via Typescript which we've done here